config/config_loader.py

```
@@ -128,4 +134,133 @@ def save_config_file(file_path: str, config: Dict[str, Any], format: Optional[st
+    
+    # Get adapter if not provided
+    if adapter is None:
+        try:
+            from security.secrets_adapter import get_adapter
+            adapter = get_adapter()
+        except ImportError:
+            # Fallback to os.environ if security module not available
+            logger.warning("security.secrets_adapter not available, using os.environ")
+            resolved = os.environ.get(secret_name)
+            if resolved is None:
+                raise ValueError(f"Secret '{secret_name}' not found in environment")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)